src/test/ui/integer-literal-suffix-inference.stderr

```diff
@@ -3,288 +3,342 @@ error[E0308]: mismatched types
    |
 LL |     id_i32(a64);
    |            ^^^ expected i32, found i64
+help: you can convert an `i64` to `i32` or panic if it the converted value wouldn't fit
+   |
+LL |     id_i32(a64.try_into().unwrap());
+   |            ^^^^^^^^^^^^^^^^^^^^^^^
 
 error[E0308]: mismatched types
   --> $DIR/integer-literal-suffix-inference.rs:64:12
    |
 LL |     id_i64(a8);
-   |            ^^ expected i64, found i8
-help: you can cast an `i8` to `i64`, which will sign-extend the source value
-   |
-LL |     id_i64(a8.into());
-   |            ^^^^^^^^^
+   |            ^^
+   |            |
+   |            expected i64, found i8
+   |            help: you can convert an `i8` to `i64`: `a8.into()`
 
 error[E0308]: mismatched types
   --> $DIR/integer-literal-suffix-inference.rs:67:12
    |
 LL |     id_i64(a16);
-   |            ^^^ expected i64, found i16
-help: you can cast an `i16` to `i64`, which will sign-extend the source value
-   |
-LL |     id_i64(a16.into());
-   |            ^^^^^^^^^^
+   |            ^^^
+   |            |
+   |            expected i64, found i16
+   |            help: you can convert an `i16` to `i64`: `a16.into()`
 
 error[E0308]: mismatched types
   --> $DIR/integer-literal-suffix-inference.rs:70:12
    |
 LL |     id_i64(a32);
-   |            ^^^ expected i64, found i32
-help: you can cast an `i32` to `i64`, which will sign-extend the source value
-   |
-LL |     id_i64(a32.into());
-   |            ^^^^^^^^^^
+   |            ^^^
+   |            |
+   |            expected i64, found i32
+   |            help: you can convert an `i32` to `i64`: `a32.into()`
 
 error[E0308]: mismatched types
   --> $DIR/integer-literal-suffix-inference.rs:76:11
    |
 LL |     id_i8(c16);
    |           ^^^ expected i8, found i16
+help: you can convert an `i16` to `i8` or panic if it the converted value wouldn't fit
+   |
+LL |     id_i8(c16.try_into().unwrap());
+   |           ^^^^^^^^^^^^^^^^^^^^^^^
 
 error[E0308]: mismatched types
   --> $DIR/integer-literal-suffix-inference.rs:79:11
    |
 LL |     id_i8(c32);
    |           ^^^ expected i8, found i32
+help: you can convert an `i32` to `i8` or panic if it the converted value wouldn't fit
+   |
+LL |     id_i8(c32.try_into().unwrap());
+   |           ^^^^^^^^^^^^^^^^^^^^^^^
 
 error[E0308]: mismatched types
   --> $DIR/integer-literal-suffix-inference.rs:82:11
    |
 LL |     id_i8(c64);
    |           ^^^ expected i8, found i64
+help: you can convert an `i64` to `i8` or panic if it the converted value wouldn't fit
+   |
+LL |     id_i8(c64.try_into().unwrap());
+   |           ^^^^^^^^^^^^^^^^^^^^^^^
 
 error[E0308]: mismatched types
   --> $DIR/integer-literal-suffix-inference.rs:86:12
    |
 LL |     id_i16(c8);
-   |            ^^ expected i16, found i8
-help: you can cast an `i8` to `i16`, which will sign-extend the source value
-   |
-LL |     id_i16(c8.into());
-   |            ^^^^^^^^^
+   |            ^^
+   |            |
+   |            expected i16, found i8
```

Comment:
I wonder if we should change the `mismathced types` root message to `expected i16, found i8` so the suggestion can be inlined properly

Reply:
It took me a second to understand but now I see what you mean... That's probably something worthy of doing in a separate PR. The actual code change should be smallish, but the effect in `stderr` files will probably be extensive.